### PR TITLE
new: Add ability to get & set grid scroll from python

### PR DESCRIPTION
**Pull request by motcke**

### CODE REVIEW

File: src/widget_builder.js [hunk 1]
```diff
@@ -226,6 +227,26 @@ const buildAgGrid = (view, gridData, gridOptions_str, div, sheet, dropdownMulti
...
+    view.model.on('change:scroll', () => {
+        let scroll = view.model.get('scroll');
+        let gridBody = view.gridDiv.querySelector("div[ref=eBodyViewport]");
+        gridBody.scrollLeft = scroll[0]
+        gridBody.scrollTop = scroll[1]
+    });
+
+
+    const saveScroll = Helpers.debounce((left, top) => {
+        console.log(`bodyScroll(${left}, ${top})`);
```
Comment (ofirbit) [inline src/widget_builder.js:241]: remove console log in line 241

File: src/widget_builder.js [hunk 2]
```diff
@@ -226,6 +227,26 @@ const buildAgGrid = (view, gridData, gridOptions_str, div, sheet, dropdownMulti
...
+        view.model.set('scroll', [left, top]);
+        view.touch();
+    }, 300);
+
+    // Listen to scroll changes from the UI
+    gridOptions.api.addEventListener('bodyScroll', params => {
+        saveScroll(params.left, params.top);
```
Comment (ofirbit) [inline src/widget_builder.js:248]: saveScroll - can you find a better name in their convention?

File: src/widget_builder.js [hunk 3]
```diff
@@ -226,6 +227,26 @@ const buildAgGrid = (view, gridData, gridOptions_str, div, sheet, dropdownMulti
...
+        gridBody.scrollLeft = scroll[0]
+        gridBody.scrollTop = scroll[1]
+    });
+
+
+    const saveScroll = Helpers.debounce((left, top) => {
+        console.log(`bodyScroll(${left}, ${top})`);
+        view.model.set('scroll', [left, top]);
+        view.touch();
+    }, 300);
```
Comment (ofirbit) [inline src/widget_builder.js:244]: add a comment next to the debounce of 300 - something like "if the user scrolls for 10 seconds, we will trigger the event only when he finished"

File: src/widget_grid.js [hunk 4]
```diff
@@ -119,9 +121,9 @@ const AgGridView = widgets.DOMWidgetView.extend({
...
             var i = sheet.rules.length
             while (i--) {
-                if (sheet.rules[i].selectorText.startsWith(`#widget-grid-${this._id}`)) { 
+                if (sheet.rules[i].selectorText.startsWith(`#widget-grid-${this._id}`)) {
                     sheet.removeRule(i);
-                } 
+                }
             }
             // Add custom CSS
             this.model.get('css_rules_down').forEach(rule => {
```
Comment (ofirbit) [inline src/widget_grid.js:129]: Let's try to:
1. Move all backend to this repository and the frontend will be inside a "js" folder. Make sure you can install the frontend and the backend from this repo
2. Add the backend change to support the scrolling.

File: src/widget_builder.js (hunk 1 above)
Comment (motcke) [reply src/widget_builder.js:241]: done

File: src/widget_builder.js (hunk 2 above)
Comment (motcke) [reply src/widget_builder.js:248]: the only convention I could find was to use the word `export` when setting property for the backend.
changed to `exportScroll`.

File: src/widget_builder.js (hunk 3 above)
Comment (motcke) [reply src/widget_builder.js:244]: done

File: src/widget_grid.js (hunk 4 above)
Comment (motcke) [reply src/widget_grid.js:129]: 1. no luck
2. done in another repo

**Pull request merged**